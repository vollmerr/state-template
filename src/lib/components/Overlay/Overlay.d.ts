import * as React from 'react';

export interface OverlayProps {
  /** Display the overlay */
  show: boolean;

  /** Click handler for overlay */
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;

  /** Key Press handler for overlay */
  onKeyPress?: (event: React.MouseEvent<HTMLButtonElement>) => void;

  /** If should only render in mobile viewport */
  isMobile?: boolean;

  /** Class name to attach */
  className?: string;
}

declare class Overlay extends React.Component<OverlayProps, {}> {}

export default Overlay;
