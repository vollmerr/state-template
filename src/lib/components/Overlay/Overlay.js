import React from 'react';
import T from 'prop-types';
import classNames from 'classnames';

/**
 * Overlays render over children content,
 * blocking any user interaction.
 */
const Overlay = (props) => {
  const {
    show,
    onClick,
    onKeyPress,
    isMobile,
    className,
    ...rest
  } = props;

  if (!show) {
    return null;
  }

  const overlayProps = {
    ...rest,
    onClick,
    role: 'presentation',
    onKeyPress: onKeyPress || onClick,
    className: classNames([
      'overlay',
      className,
      { 'mobile-only': isMobile },
    ]),
  };

  return <div {...overlayProps} />;
};

Overlay.propTypes = {
  /** Display the overlay */
  show: T.bool.isRequired,
  /** Click handler for overlay */
  onClick: T.func,
  /** Key Press handler for overlay */
  onKeyPress: T.func,
  /** If should only render in mobile viewport */
  isMobile: T.bool,
  /** Class name to attach */
  className: T.string,
};

Overlay.defaultProps = {
  onClick: undefined,
  onKeyPress: undefined,
  isMobile: false,
  className: '',
};

export default Overlay;
