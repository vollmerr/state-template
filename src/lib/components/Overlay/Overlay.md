```jsx
import { Button } from 'state-template';

class ExampleOverlay extends React.Component {
  constructor(props) {
    super(props);
    this.state = { isOpen: false }
    this.toggleOpen = this.toggleOpen.bind(this);
  }

  // use arrow syntax here instead of binding above...
  // limitation of styleguidist transpiler..
  toggleOpen() {
    const { isOpen } = this.state;
    this.setState({ isOpen: !isOpen });
  }

  render() {
    const { isOpen } = this.state;
    
    return (
      <div>
        <Button onClick={this.toggleOpen} text={'open'} variant={'default'} />
        <Overlay onClick={this.toggleOpen} show={isOpen} />
      </div>
    );
  }
}

<ExampleOverlay />
```
