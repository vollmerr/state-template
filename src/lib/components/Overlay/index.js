import React from 'react';
import T from 'prop-types';

/**
 * Renders an overlay that blocks user interaction
 */
const Overlay = (props) => {
  const {
    show,
    onClick,
    onKeyPress,
    isMobile,
    ...rest
  } = props;

  if (!show) {
    return null;
  }

  const overlayProps = {
    onClick,
    role: 'presentation',
    onKeyPress: onKeyPress || onClick,
    className: `overlay ${isMobile && 'mobile-only'}`,
    ...rest,
  };

  return <div {...overlayProps} />;
};

Overlay.propTypes = {
  /** Display the overlay */
  show: T.bool.isRequired,
  /** Click handler for overlay */
  onClick: T.func,
  /** Key Press handler for overlay */
  onKeyPress: T.func,
  /** If should only render in mobile viewport */
  isMobile: T.bool,
};

Overlay.defaultProps = {
  onClick: undefined,
  onKeyPress: undefined,
  isMobile: false,
};

export default Overlay;
