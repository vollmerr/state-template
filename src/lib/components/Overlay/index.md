```jsx
class ExampleOverlay extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = { isOpen: false }
    this.toggleOpen = this.toggleOpen.bind(this);
  }

  // use arrow syntax here instead of binding above...
  // limitation of styleguidist transpiler..
  toggleOpen() {
    const { isOpen } = this.state;
    this.setState({ isOpen: !isOpen });
  }

  render() {
    const { isOpen } = this.state;
    
    return (
      <div>
        <Button onClick={this.toggleOpen} text={'open'} />
        <Overlay onClick={this.toggleOpen} show={isOpen} />
      </div>
    );
  }
}

<ExampleOverlay />
```
