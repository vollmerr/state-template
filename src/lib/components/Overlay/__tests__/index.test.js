import React from 'react';
import { shallow } from 'enzyme';

import { Overlay as fromLib } from '../../..';
import Overlay from '..';

const props = {
  show: true,
  onClick: jest.fn(),
};

let wrapper;
describe('Overlay', () => {
  beforeEach(() => {
    wrapper = shallow(<Overlay {...props} />);
  });

  it('should be exported as part of the library', () => {
    expect(fromLib).toEqual(Overlay);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
