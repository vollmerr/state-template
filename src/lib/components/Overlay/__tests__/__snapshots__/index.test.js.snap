// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay should render correctly 1`] = `
<div
  className="overlay"
  onClick={[MockFunction]}
  onKeyPress={[MockFunction]}
  role="presentation"
/>
`;
